pkg/kubelet/kubelet.go

```diff
@@ -1536,6 +1537,18 @@ func (kl *Kubelet) GetPodByName(namespace, name string) (*api.BoundPod, bool) {
 	return nil, false
 }
 
+// getPodByUID returns the pod that matches 'namespace' and 'uid', nil if no pod matches.
```

Comment:
UID is unique in time and space crossing all namespaces. thus there is no need to get pod by UID and namespace.